src/http/web_socket.cr

```diff
@@ -101,86 +101,109 @@ class HTTP::WebSocket
     @ws.close(message)
   end
 
-  record Ping, message : String
-  record Pong, message : String
-  record Text, message : String
-  record Binary, message : Bytes
-  record Close, message : String
-
-  def handle_opcodes(data)
-    case data
-    when Ping
-      pong(data.message) unless closed?
-      return nil
-    when Pong
-      return nil
-    when Close
-      close(data.message) unless closed?
-      @close_reason = data.message
-      return nil
+  enum Opcode
+    Text
+    Binary
+    Close
+    Ping
+    Pong
+  end
+
+  struct Message
+    getter opcode
+
+    def text
+      @data.as(String)
     end
 
-    data
+    def binary
+      @data.as(Bytes)
+    end
+
+    def initialize(@opcode : Opcode, @data : String | Bytes)
+    end
+  end
+
+  private def handle_opcodes(message)
+    case message.opcode
+    when Opcode::Close
+      close(message.text) unless closed?
+      @close_reason = message.text
+    when Opcode::Ping
+      pong(message.text) unless closed?
+    end
+
+    if message.is_a?(String | Bytes)
```

Comment:
This will always fail.

Reply:
Oops, forgot to change this when I changed the message to a struct.